appserver/batch/glassfish-batch-connector/src/main/java/fish/payara/jbatch/persistence/rdbms/MySqlPersistenceManager.java

```diff
@@ -272,11 +272,13 @@ protected void createMySQLTableNotExists(String tableName,
     }
 
     @Override
-    protected void setSchemaOnConnection(Connection connection) throws SQLException {
-            PreparedStatement ps = null;
-            ps = connection.prepareStatement("USE " + schema);
-            ps.executeUpdate();
-            ps.close();
+    protected void setSchemaOnConnection(Connection connection) throws SQLException {      
+            try (PreparedStatement preparedStatement = connection.prepareStatement("USE " + schema)) {
+                preparedStatement.executeUpdate();
+
+            } catch (SQLException ex) {
+                logger.severe(ex.getLocalizedMessage());
```

Comment:
@MeroRai Should you catch the SQLException here? I think the previous implementation would have thrown an SQLException instead. At least the method signature suggests this.

Reply:
@svendiedrichsen I must have overlooked it. You are alright, it should throw the exception.